SceneRefAttributeValidator.cs

```diff
@@ -319,44 +354,92 @@ object existingValue
                 iSerializable?.Clear();
 #if UNITY_EDITOR
                 if (existingValue != null)
-                    EditorUtility.SetDirty(c);
+                {
+                    EditorUtility.SetDirty(component);
+                }
 #endif
                 return null;
             }
 
-            if (iSerializable != null)
+            if (iSerializable == null)
             {
-                if (!iSerializable.OnSerialize(value))
+                bool valuesAreEqual = existingValue != null && (isArray ? ((IEnumerable)value).HaveSameCount(existingValueEnumerable) : value.Equals(existingValue));
```

Comment:
Checking they have the same count isn't enough to measure equality, we need to maintain the element-wise equality check of `Enumerable.SequenceEqual`

Reply:
Added it back in